transportmiddleware/interface.go

```diff
@@ -0,0 +1,53 @@
+// Package transportmiddleware provides an interface for hooking into
+// outgoing yab requests.
+package transportmiddleware
```

Comment:
is it worth having a special interface for this? does this make more sense as `transport.Middleware`